README.md

```diff
@@ -19,6 +19,23 @@ if hits:
     print(hits.reason_codes)
 ```
 
+Auth
+----
+
+If the target Checkmate server requires authentication 
+use the provided API key defining a environment variable:
+
+```
+CHECKMATE_API_KEY=YOUR_KEY
+```
+
+or explicitly in your code when instantiating the client:
+
+```python
+client = CheckmateClient("http://checkmate.example.com", api_key="YOUR_KEY")
+```
```

Comment:
I think we should just make the environment variable mandatory and integrate this into the **Usage** section above -- we'll just make Checkmate's API require an API key (mandatory)

Reply:
> I think we should just make the environment variable mandatory and integrate this into the **Usage** section above -- we'll just make Checkmate's API require an API key (mandatory)

I could remove the option to pass the api_key to the CheckmateClient() an rely only on the environment variable but making it mandatory (eg doing os.environ[] and failing if not there) could make our lives more complicated to deploy this across services (well, actually only changing the order, first we need the variable, then the updated library, not the end of the world)

Is failing if the env var is not set what you meant?

Reply:
Yeah I'd just do `os.environ[...]` and let it `KeyError`.

We'd have to add the envvar to each app's environment _before_ updating checkmatelib in each app.

I guess the API key is never _really_ going to be optional, as I don't think we're going to make authentication optional in checkmate. So even if checkmatelib allows you to use its client without an API key, checkmatelib's requests to checkmate would then fail.

I don't think it matters too much either way though--having the API key be optional in checkmatelib is fine.

Reply:
Make sense, done that on the last commit